src/Chapter1.hs

```diff
@@ -542,7 +543,12 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+   | (x < y && y < z) || (z < y && y < x) = y
+   | (y < x && x < z) || (z < x && x < y) = x
+   | (x == y) || (y == z)                 = y
```

Comment:
That is the correct holution!
However, you can move this case under two previous ones by using `<=` instead of just `<`:
```haskell
   | (x <= y && y <= z) || (z <= y && y <= x) = y
```